job/dhcp.go

```diff
@@ -175,5 +174,10 @@ func (j Job) setPXEFilename(rep *dhcp4.Packet, isTinkerbellIPXE, isARM, isUEFI,
 		return
 	}
 
+	// If the client speaks HTTP, use the PublicFQDN.
+	if isHTTPClient {
```

Comment:
hmmm....this might not need to be wrapped in an `if` statement at all. line 182 (`dhcp.SetFilename`) does this same check. https://github.com/tinkerbell/boots/blob/fdd7a4fb385659328bb8c3c6625741936c00aaa5/dhcp/pxe.go#L143

Reply:
one option could be to just use the previous code, like you referenced in [this](https://github.com/tinkerbell/boots/pull/229#discussion_r836704325) comment.

Reply:
At that rate, at least for now, I could just modify the function call directly:
```diff
- dhcp.SetFilename(rep, filename, j.NextServer, isHTTPClient, j.HTTPServerFQDN)
+ dhcp.SetFilename(rep, filename, j.NextServer, isHTTPClient, conf.PublicFQDN)
```

Reply:
Changed.

Reply:
I imagine there was a purpose behind that other change which in the future we should untangle and move forward with at some point, but for now I think this will fix our production issue.